dotCMS/src/main/java/com/dotmarketing/business/portal/PortletAPIImpl.java

```diff
@@ -105,43 +110,29 @@ public void deletePortlet(String portletId) {
 
     @Override
     @WrapInTransaction
-    public Portlet savePortlet(final Portlet portlet, final User user) throws DotDataException, LanguageException {
-
-        if(!UtilMethods.isSet(portlet.getPortletId())) {
-            throw new DotDataValidationException("Portlet Id is Required");
-        }
-        final String portletId = CONTENT_PORTLET_PREFIX + portlet.getPortletId();
-        if (UtilMethods.isSet(findPortlet(portletId))) {
-            throw new DotDataValidationException("Portlet Id already Exists");
-        }
-        if(!UtilMethods.isSet(portlet.getInitParams().get("name"))) {
-            throw new DotDataValidationException("Portlet Name is Required");
-        }
-        if(!UtilMethods.isSet(portlet.getPortletClass())) {
-            throw new DotDataValidationException("You cannot save a portlet without an implementing portletClass");
-        }
+    public Portlet createOrUpdatePortlet(Portlet portlet, User user) throws DotDataException, LanguageException {
```

Comment:
Let's keep the same name, just in case a customer is calling this method in a custom plugin.

Also, let's add final to the params.

Reply:
Done https://github.com/dotCMS/core/pull/27986/commits/8f9eef068702ebb83612e9044280e1ea9f9ddd10